swift-extensions/UIButtonExtensions.swift

```diff
@@ -199,15 +176,15 @@ extension UIButton {
     }
 
     func updateBackgroundColor() {
-        guard let metaButton = metaButton else { return }
-        observe(metaButton.backgroundColor.first()) {[weak self] (selector: MetaSelector) in
+        guard let buttonViewModel = buttonViewModel else { return }
+        observe(buttonViewModel.backgroundColor.first()) {[weak self] (selector: StateSelector) in
             self?.updateBackgroundColor(selector)
         }
     }
 
     @objc
     private func onButtonTouchUp() {
-        guard let tapAction = tapAction else { return }
-        tapAction.execute(actionContext: self ?? nil)
+        guard let buttonViewModel = buttonViewModel else { return }
+        observe(buttonViewModel.action.first()) {[weak self] (value: ViewModelAction) in value.execute(actionContext: self ?? nil) }
```

Comment:
T'as enlevé la modif de Marc? Le first est un problème à cause de l'operator startWith qui contient un buffer de value dans le publisher non.

Reply:
I'll fix it!